Lift more info to Supported environments page (#3310)
camunda/camunda-docs: docs/reference/supported-environments.md

description: "Find out where to run Camunda 8 components for SaaS and Self-Managed, including Optimize for both Camunda 8 and Camunda 7."
---

> The supported environments page lists tested and compatible environments, deployment options, and component dependencies for successfully running Camunda 8.

Reviewer: This sentence reads strange to me. What about?

```suggestion
The supported environments page lists environments, deployment options, and component dependencies, which are tested and supported for compatibility with Camunda 8.
```

Reviewer 2: either way is fine for me

> **If the particular technology is not listed, we are unable to resolve issues or provide guidance on integration.** You may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda.

Reviewer: Should we re-use the wording from bevore for clarity?
```suggestion
**If you are running Camunda 8 using an environment, deployment option, or component dependency that is not listed, we cannot resolve issues or provide guidance on integration.** You may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda.
```

Reviewer 2: Make sense from a consistency, but do we define environment, deployment option or component dependency anywhere? A customer will what to know what we mean. If we keep it as 'technology' we can use this a catch all which would be my preference instead of having to maintain more definitions

Reviewer: The idea of this is to re-use the wording that is part of the page. The page would thereby give the definition. E.g., 

`Deployment Option`=> What a deployment option is, is defined as by the sub section of this page, which talks about `Deployment Options: HELM, Docker-compose, ...`.

@akeller can you please try to adjust this statement so that we pick up the terms of the sub-sections?

Author: "The supported environments page lists browsers, operating systems, clients, deployment options, and component requirements, which are tested and supported for compatibility with Camunda 8."

> Recommendations are denoted with [recommended], however, other options are supported as well.

Reviewer: Maybe more verbose to ensure it is clearly understood what we mean:

```suggestion
If you started without any technology contraints, we advise you to follow our recommendations, which are denoted with [recommended]. However, the other listed options are supported as well.
```

Author: I like adding "the other listed options are supported as well," but I'm not sure we need the first half of the sentence to specify recommendations are for those without technology constraints. 

Since we use recommendations in other areas of our docs, we can add this as a glossary term and commit to what recommendations mean in the scope of Camunda docs.

Reviewer: Okay. Can you please commit this proposal?

description: "Find out where to run Camunda 8 components for SaaS and Self-Managed, including Optimize for both Camunda 8 and Camunda 7."
---

The supported environments page lists browsers, operating systems, clients, deployment options, and component requirements, which are tested and supported for compatibility with Camunda 8.

> **If the particular technology is not listed, we are unable to resolve issues or provide guidance on integration.** You may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda.

Reviewer: Align with latest discussions in https://docs.google.com/document/d/1PjxMqicBKNCll-7EyJCga7n3NkWWMeyXgwfdyZ3D7eU/edit

Proposal:
```suggestion
**If the particular technology is not listed, it may cause issues for Camunda, which we cannot resolve.** If you rely on a technology that is not supported, you may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda, or [raise a help request](/contact) to request Consulting services.
```

Author: Let's just keep this brief and more open-ended - if the particular technology is not listed, and you're looking for support OR guidance (like Best Practices), we wouldn't be able to help outside of a consulting request. 

```suggestion
**If the particular technology is not listed, we cannot resolve issues or guide integrations.** 

You may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda, or you can make a [help request](/contact) to work with Consulting services. 
```

I'm hesitant to add "guide integrations" here because we offer possible guidance through Consulting, but it's up to their discretion. I'd like to see this clearly defined in our contracts because the way it currently reads, it sounds like Consulting will support anything and everything via help requests, and that may not be true.

Author: ```suggestion
**If the particular technology is not listed, we cannot resolve issues caused by the usage of that unlisted technology.**

You may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda, or you can make a [help request](/contact) to work with Consulting services. 
```

Reviewer: Agreed. Let's do that.

## Camunda 8 Self-Managed

### Minimum requirements

The following are the minimum supported requirements for cluster specification and dependencies:

> - 4vCPUs, 15 GB Memory

Reviewer 3: small note. vCPU will not be applicable on bare-metal installs. maybe we can call our "4 (v)CPU"

Author: Sure! @theburi Do we have documentation on bare-metal installs? Would this be [manual installation?](https://docs.camunda.io/docs/self-managed/platform-deployment/manual/)

Reviewer 3: Generally, we would expect users to use Manual installation, but they could also use Docker-compose (Docker standalone). Looking at my comment today, CPU to vCPU relationship is hard to describe as there are different calculations. So I think it becomes too nuanced and we should not go there.

- 4 nodes
- OpenJDK 17+
- Elasticsearch 8.9+ or Amazon OpenSearch 2.5+
- PostgreSQL 14.x or Amazon Aurora PostgreSQL 14.x
- ...

> ### Recommendations

Reviewer 3: Should we add a section for none-Kubernetes deployments? 
We still support Docker standalone (Compose) and Manual, which could be deployed on IaaS Cloud, VMs or Bare-metal.

Author: My understanding is that Docker Compose is not for production use, and I would not highlight it here. This should be production only. Please confirm you agree with the split [here](https://docs.camunda.io/docs/self-managed/platform-deployment/overview/#deployment-recommendation).

We've previously been very explicit about Helm as our recommended deployment path. Should we remove this and promote all deployment methods? This would be a change in direction in how we've presented this content.

Reviewer 3: We officially support [4 production configurations](https://docs.camunda.io/docs/self-managed/platform-deployment/overview/#production) 
Let me clarify that we don't support docker-compose 'script' that deploys C8, but we support C8 deployed with Docker images (Which in reality would be done with customer-provided scripts using docker-compose similar to customer providing script to deploy JAR's)
We highly recommend HELM and Kubernetes, but our customers deploy with other options as well and we support them.

## Camunda 8 Self-Managed

> We highly recommend running Camunda 8 Self-Managed in a Kubernetes environment. We provide officially supported [Helm Charts](/self-managed/platform-deployment/helm-kubernetes/overview.md) for this. Follow the [Installation Guide](/self-managed/platform-deployment/overview.md) to learn more about installation possibilities.

Reviewer: Should we explicitly link and state here which setups we support / not support? Basically what we have here: https://docs.camunda.io/docs/next/self-managed/platform-deployment/overview/#deployment-options

Author: I added links to the deployment options in my latest commit. We shouldn't document what we don't support. We should focus on what we do support.

> The following list includes links for supported Kubernetes environments and recommended starting config. Generally speaking, the specification depends on your needs and workloads.

Reviewer: @akeller can you please make this more precise (or align with EMs to do so)? Is this the absolute minium requirement you always must have, or is the the "average", but you can be above or below based on workloads?

Author: AFAIK, this is average. Minimum would imply a smaller installation footprint (1 Zeebe vs. 3) and possible reduced availability.

> - [Amazon EKS](/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/amazon-eks.md)

Author: Yes, we have some ways to manage redundancy with reusable comments. I added "
Identify content to move to reusable components" to the [issue tasklist](https://github.com/camunda/camunda-docs/issues/3306).

This page would be too big if we put all the information on a single page for all our supported environments.

  - Instance type: `m6i.xlarge` (4 vCPUs, 16 GiB Memory)
  - Number of nodes: `4`
  - Volume type: `SSD gp3`
    - Including Amazon EBS CSI driver and StorageClass
- [Microsoft AKS](/self-managed/platform-deployment/helm-kubernetes/platforms/microsoft-aks.md)
  - Instance type: `Standard_D4as_v4` (4 vCPUs, 16 GiB Memory)
  - Number of nodes: `4`
  - Volume type: `Premium SSD/Premium SSD v2`
- [Google GKE](/self-managed/platform-deployment/helm-kubernetes/platforms/google-gke.md)
  - Instance type: `n1-standard-4` (4 vCPUs, 15 GB Memory)
  - Number of nodes: `4`
  - Volume type: `Performance (SSD) persistent disks`
- [Red Hat OpenShift](/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md) - 4.11.x, 4.12.x, 4.13.x

> <!-- No current documentation available for recommended starting config.-->

Author: Are there starting config recommendations for Red Hat? Currently, it's the only platform without them, but with specific supported versions.

Reviewer 3: OpenShift does not have associated instance sizes as it could be installed on any infrastructure.

Author: I will adjust this so the expectations are clear and it doesn't look like a miss.

### Sizing

The [sizing of a Camunda 8 installation](/components/best-practices/architecture/sizing-your-environment.md) depends on various influencing factors. Ensure to [determine these factors](../components/best-practices/architecture/sizing-your-environment.md#understanding-influencing-factors), and conduct [benchmarking](../components/best-practices/architecture/sizing-your-environment.md#running-experiments-and-benchmarks) to validate an appropriate environment size for your test, integration, or production environments.

#### Volume Performance

As a minimum requirement the cluster nodes should use volumes with an absolute minimum of 1.000 IOPS. **NFS volumes are not supported.** To ensure an appropriate sizing, ensure to [determine your influencing factors](../components/best-practices/architecture/sizing-your-environment.md#understanding-influencing-factors) (e.g., throughput), and conduct [benchmarking to validate an appropriate environment sizing](../components/best-practices/architecture/sizing-your-environment.md#running-experiments-and-benchmarks).

> Refer to the following Cloud-provider specific examples, for typical volume types to be used.
> 
> Depending on your cloud provider, we offer the following volume performance recommendations:

Reviewer: ```suggestion
Refer to the following Cloud-provider specific examples, for typical volume types to be used:

```

## Camunda 8 Self-Managed

### Deployment options

- [Helm/Kubernetes](/self-managed/platform-deployment/helm-kubernetes/overview.md) [recommended]
  - [Platforms](/self-managed/platform-deployment/helm-kubernetes/platforms/platforms.md)
    - [Amazon EKS](/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/amazon-eks.md)
    - [Microsoft AKS](/self-managed/platform-deployment/helm-kubernetes/platforms/microsoft-aks.md)
    - [Google GKE](/self-managed/platform-deployment/helm-kubernetes/platforms/google-gke.md)
    - [Red Hat OpenShift](/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md) (4.11+)
- [Docker](/self-managed/platform-deployment/docker.md)
- [Manual](/self-managed/platform-deployment/manual.md)

> Ensure the Camunda component versions are compatible with the Helm chart version as defined in the [matrix](https://helm.camunda.io/camunda-platform/version-matrix/).

Reviewer: This statement is only valid for the "HELM" option, yet it reads and is placed as if it would apply to the entire section. Can you please check whether we cannot put it under the - HELM list point?

Author: I moved this to an admonition titled "Helm chart compatibility".

Adding it to the list would make for a disjointed list.

## Camunda 8 Self-Managed

### Deployment options

With the right configuration, Camunda 8 Self-Managed can be deployed on any [Certified Kubernetes](https://www.cncf.io/training/certification/software-conformance/#benefits) distribution (cloud or on-premise). However, we officially test and support a specific list of platforms.

The following are tested and supported deployment options for Kubernetes, Docker, and manual installation:

> - [Stock Kubernetes](/self-managed/platform-deployment/helm-kubernetes/overview.md)

Reviewer 3: We need to clarify what does it mean -Stock Kubernetes. It is not a standard term that used in CNCF. Does it mean upstream or certified?
imo we should support any certified distribution and specifically highlight tested platforms. (In a similar fashion to Elastic or Confluence)

Reviewer 4: This is a great callout. I think this should likely just be referenced as Kubernetes (k8s).

We can add style guide assistance on this once agreed upon here.

Reviewer 3: ok. Let's keep it as is, so we can progress here, and I will clarify and make another Epic to clarify this specific point and we will make changes throughout the docs.

  - [Microsoft AKS](/self-managed/platform-deployment/helm-kubernetes/platforms/microsoft-aks.md)
  - [Google GKE](/self-managed/platform-deployment/helm-kubernetes/platforms/google-gke.md)
- [Red Hat OpenShift](/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md) (4.11+)
- [Docker](/self-managed/platform-deployment/docker.md)
- [Manual](/self-managed/platform-deployment/manual.md)

:::note Helm chart compatibility
Ensure the Camunda component versions are compatible with the Helm chart version as defined in the [matrix](https://helm.camunda.io/camunda-platform/version-matrix/).
:::

### Sizing

The [sizing of a Camunda 8 installation](/components/best-practices/architecture/sizing-your-environment.md) depends on various influencing factors. Ensure to [determine these factors](../components/best-practices/architecture/sizing-your-environment.md#understanding-influencing-factors), and conduct [benchmarking](../components/best-practices/architecture/sizing-your-environment.md#running-experiments-and-benchmarks) to validate an appropriate environment size for your test, integration, or production environments.

#### Volume performance

> As a minimum requirement the cluster nodes should use volumes with an absolute minimum of 1.000 IOPS. **NFS volumes are not supported.**

Reviewer 5: It recently came up whether we support other network storage types, I would expand on stating no network storage is supported.

```suggestion
As a minimum requirement the cluster nodes should use volumes with an absolute minimum of 1.000 IOPS. **NFS or other types of network storage volumes are not supported.**
```

Author: @megglos I will also follow up in a separate PR to update this on other pages.

Camunda 8 works with the [default distribution](https://www.elastic.co/downloads/elasticsearch) of Elasticsearch, which is available under the [Free or Gold+ Elastic license](https://www.elastic.co/pricing/faq/licensing#summary).
:::

### Volume Performance

> Depending on your cloud provider, we offer the following volume performance recommendations:

Reviewer: Also here, can we be specific please (recommendation vs minium requirement)? I would love the see minium requirements.

Reviewer 3: We should define volume performance parameters to cover on-premises or "exotic cloud"  (i.e. Oracle, IBM, Digital Ocean, hosting providers)

Reviewer 3: We come across questions about the support of NFS. In some environments, NFS could perform as well as local storage, and we need to define required performance characteristics and maybe call out some NFS's that we know are not supported.

Author: @theburi I view both of these items as feature enhancements and outside of the scope of this PR. Can you create epics to drive these changes with the appropriate team (Distribution or InfraEx)?

Reviewer 3: Yes. Let's add this reference.

- [Amazon EKS](/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks#volume-performance)
- [Google GKE](/self-managed/platform-deployment/helm-kubernetes/platforms/google-gke#volume-performance)
- [Microsoft AKS](/self-managed/platform-deployment/helm-kubernetes/platforms/microsoft-aks#volume-performance)

> - [Red Hat OpenShift](#) <!-- No current documentation available. -->

Author: OpenShift has no docs on volume performance. Do we need to add them?

### Helm chart version matrix

The core Camunda components have a unified fixed release schedule following the [release policy](./release-policy.md). However, some of the applications have their own schedule. The following compatibility matrix gives an overview of the different versions with respect to the Helm chart versions.

Since the 8.4 release (January 2024), the Camunda Helm chart version is decoupled from the version of the application (e.g., the chart version is 9.0.0 and the application version is 8.4.x).

> <!-- Nit pick - this is a list, not a matrix. This is also on a separate site with no oversight from DevEx. -->

Author: @MaxTru we should have a discussion about either collapsing these into docs.camunda.io or getting DevEx involved in reviews for helm.camunda.io.

### Applications version matrix

This overview shows which Zeebe version works with which Modeler, Operate, Tasklist, and Optimize:

| Design                | Automate    |                                             | Improve         |
| --------------------- | ----------- | ------------------------------------------- | --------------- |
| Desktop Modeler 4.12+ | Zeebe 1.3.x | Operate 1.3.x Tasklist 1.3.x IAM 1.3.x      | Optimize 3.7.x  |
| Desktop Modeler 5.0+  | Zeebe 8.0.x | Operate 8.0.x Tasklist 8.0.x Identity 8.0.x | Optimize 3.8.x  |
| Desktop Modeler 5.4+  | Zeebe 8.1.x | Operate 8.1.x Tasklist 8.1.x Identity 8.1.x | Optimize 3.9.x  |
| Desktop Modeler 5.10+ | Zeebe 8.2.x | Operate 8.2.x Tasklist 8.2.x Identity 8.2.x | Optimize 3.10.x |
| Web Modeler 8.2.x     | Zeebe 8.2.x | Operate 8.2.x Tasklist 8.2.x Identity 8.2.x | Optimize 3.10.x |
| Web Modeler 8.3.x     | Zeebe 8.3.x | Operate 8.3.x Tasklist 8.3.x Identity 8.3.x | Optimize 8.3.x  |
| Web Modeler 8.4.x     | Zeebe 8.4.x | Operate 8.4.x Tasklist 8.4.x Identity 8.4.x | Optimize 8.4.x  |

:::note
You can also use newer versions of Desktop and Web Modeler with older Zeebe versions.
:::

> ## form-js version matrix

Reviewer: @akeller why is this needed from a customer perspective? Where is the source of this table?

Author: https://github.com/camunda/camunda-docs/pull/3300

This is the source. I merged it as a quick win, but we can move it elsewhere. I didn't have a better recommendation of where to put it.

| Identity    | OpenJDK 17+  | Keycloak 22.x, 23.x<br/>PostgreSQL 14.x, 15.x (required for [certain features](/self-managed/identity/deployment/configuration-variables.md#database-configuration)) |
| Optimize    | OpenJDK 17+  | Elasticsearch 8.9+                                                                                                                                                   |
| Web Modeler | -            | PostgreSQL 13.x, 14.x, 15.x, 16.x or Amazon Aurora PostgreSQL 13.x, 14.x, 15.x, 16.x                                                                                 |

When running Elasticsearch, you must have the [privileges listed here](/self-managed/concepts/elasticsearch-privileges.md).

:::note Elasticsearch support
Camunda 8 works with the [default distribution](https://www.elastic.co/downloads/elasticsearch) of Elasticsearch, which is available under the [Free or Gold+ Elastic license](https://www.elastic.co/pricing/faq/licensing#summary).
:::

### Component version matrix

> This overview shows which Zeebe version works with which Modeler, Operate, Tasklist, and Optimize:

Reviewer: ```suggestion
This overview shows which Zeebe version works with which Modeler, Operate, Tasklist, and Optimize version:
```

Author: This sentence is still a little awkward. @christinaausley what would you do here?
